java/src/main/java/org/signal/libsignal/metadata/encoding/Base32.java

```diff
@@ -0,0 +1,140 @@
+package org.signal.libsignal.metadata.encoding;
+
+import java.util.Map;
+
+/**
+ * An implementation of Base32 encoding/decoding that accepts arbitrary alphabets.
+ * From http://www.herongyang.com/Encoding/Base32-Bitpedia-Java-Implementation.html;
+ *
+ * It also supports a replacement table on decoding to handle mis-entered letters.
+ */
+public class Base32 {
```

Comment:
I might eventually move this somewhere else since it's not specific to sealed-sender functionality, but I needed it here for now.